app/models/district.rb

```diff
@@ -17,11 +17,19 @@ class District < ActiveRecord::Base
   validates :state, presence: true
   validates :district, uniqueness: { scope: :state }
 
-  def to_s
-    state.abbrev + district
+  def self.find_by_hash(state:, district:)
+    District.find_by(state: State.find_by(abbrev: state), district: district)
```

Comment:
Good to break out this method, but you're generating two sql queries here, and an third by not using `includes` when campaigns gets called. 

use `.includes(:campaigns, :state).where('states.abbrev': results[:state]).find_by(district:district)` I'd also argue that find_by hash is a bit ambiguous - even though it is long, `find_by_state_and_district` would be more clear.